### ISSUE TITLE
status of this project

### ISSUE THREAD

**Issue by soederpop** [rehashstudio/react-native-draft-js#2]
Hi,

I found this project looking for a draftjs editor for react-native.   Currently I use the react-native-draftjs-renderer to display draft content created in our webapp.  I'd love to offer some basic editing capabilities.  

Is that a goal of this project?

JS

**Comment by abeck99** [comment]
Yes absolutely! However it’s iOS only and have no plans on android support.
We have already have it in a production app, and I’ve been planning on open sourcing it. 

The text rendering is done completely native with TextKit (as opposed to using a standard UIKit widget), and atomic blocks are react native components, with the limitation that width/height needs to be known before hand (this is fixable but I have no plans to address it). The native component just marshals text entry to js and displays the draftjs content, so all text manipulation is done entirely in js world.

The reason why it’s structured this way is because the input from standard text fields from iOS don’t give you all input from the keyboard, react native tries to standardize it which turns out to be leaky, and draftjs is built for web only and default modifiers assume some things about the web text input widgets. In short, it’s a mess, so doing a dumb native text input/renderer with all the “business logic” in draftjs is really the only viable approach, it just cuts through all the layers of leaky abstraction.

If you don’t care about the native details you’ll be able to do all the business logic in draftjs and not have to worry about implementation!

I’ll put something up here in the next few days and let you know

Andrew

**Comment by jpstrikesback** [comment]
Hey Andrew,

Was there an example released, or is this repo enough to get going with some grokking?

Cheers!
Jon

**Comment by abeck99** [comment]
Hi!

There's no example, the above comment is really the only "documentation" - it was more like we literally didn't find any decent solution for native draftjs editor (lots of good visualizers), so made one ourselves with native text rendering and threw it up because it's useful. Making an example is good idea and if I have some time I will do that. In the meantime, it should be possible for someone who understands RN to figure out - there are a few brittle parts (specifically how width/height is handled is not coming from yoga layouts so needs to be hardcoded) and a few non-standard things posing as standard things (the text formatting css values it not going through the full react-native css system, but it does use the react-native built in to resolve text font).

If you do jump in, I'm happy to answer any questions, and we can make the beginning of a faq/doc from that. I know the iOS part very well but we used a contractor for the Android aspect so I may not be able to answer everything about that.

Andrew

**Comment by jpstrikesback** [comment]
Hey! Thanks so much for the detailed response, super appreciated!!

I'm fairly comfortable in the RN iOS side, I plan to throw a few hours at it tonight and I'll be sure to ask questions :D & share any findings!

Thanks!
Jon
